extensions/csv/src/main/java/io/deephaven/csv/reading/CsvReader.java

```diff
@@ -0,0 +1,302 @@
+    private boolean async = false;
+    /**
+     * The user-defined set of parsers that participate in type inference. Defaults to ZamboniParsers.DEFAULT
+     */
+    private Set<ParserBase<?>> inference = new HashSet<>(Parsers.DEFAULT);
+    /**
+     * Client-specified headers that can be used to override the existing headers in the input (if hasHeaders is true),
+     * or to provide absent headers (if hasHeaders is false).
+     */
+    private List<String> clientSpecifiedHeaders = new ArrayList<>();
+    /**
+     * Used to force a specific parser for a specific column.
+     */
+    private Map<String, ParserBase<?>> parsers = new HashMap<>();
+    /**
+     * The parser
+     */
+    private ParserBase<?> nullParser;
+    /**
+     * The NullContext holding various null sentinels.
+     */
+    private final SentinelConfiguration nullContext = new SentinelConfiguration();
+
+    public CsvReader(Reader reader, SinkFactory sinkFactory) {
+        this.reader = reader;
+        this.sinkFactory = sinkFactory;
+    }
+
+    public Result read() throws IOException {
+        final CellGrabber grabber = new CellGrabber(reader, quoteChar, fieldDelimiter, ignoreSurroundingSpaces, trim);
+        String[] headersToUse = null;
+        if (hasHeaders) {
+            headersToUse = tryReadOneRow(grabber);
+            if (headersToUse == null) {
+                throw new RuntimeException("Can't proceed because hasHeaders is set but input file is empty");
+            }
+        }
+
+        // If not, maybe override with client-specified headers
+        if (clientSpecifiedHeaders.size() != 0) {
+            headersToUse = clientSpecifiedHeaders.toArray(new String[0]);
+        }
+
+        // If not, maybe generate synthetic column headers (works only if the file is non-empty)
+        String[] firstDataRow = null;
+        if (headersToUse == null) {
+            firstDataRow = tryReadOneRow(grabber);
+            if (firstDataRow == null) {
+                throw new RuntimeException(
+                        "Can't proceed because input file is empty and client has not specified headers");
+            }
+            headersToUse = new String[firstDataRow.length];
+            for (int ii = 0; ii < headersToUse.length; ++ii) {
+                headersToUse[ii] = "Column" + (ii + 1);
+            }
+        }
+
+        final int numCols = headersToUse.length;
+
+        // Create a DenseStorageWriter and two readers for each column.
+        final DenseStorageWriter[] dsws = new DenseStorageWriter[numCols];
+        final DenseStorageReader[] dsr0s = new DenseStorageReader[numCols];
+        final DenseStorageReader[] dsr1s = new DenseStorageReader[numCols];
+        for (int ii = 0; ii < numCols; ++ii) {
+            final DenseStorageWriter dsw = new DenseStorageWriter();
+            dsws[ii] = dsw;
+            dsr0s[ii] = dsw.newReader();
+            dsr1s[ii] = dsw.newReader();
+        }
+
+        // run in separate thread if async
```

Comment:
All the other comments start with a capital letter, and have a different "feel", I am guessing you made this at a different time and/or as a note to yourself.   I am not asking you to do anything here, just raising to you that this is noticeable for me as I read through the code the first time.